provider/resource_edge_deployment.go

```diff
@@ -0,0 +1,42 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceEdgeDeployment() *schema.Resource {
+	return &schema.Resource{
+		Create:   createOrUpdateEdgeDeployment,
+		Read:     readEdgeDeployment,
+		Update:   createOrUpdateEdgeDeployment,
+		Delete:   deleteEdgeDeployment,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			"site_short_name": {
+				Type:        schema.TypeString,
+				Description: "Site short name",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func createOrUpdateEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	pm := m.(providerMetadata)
```

Comment:
what happens when m is not a providerMetadata? Is that possible?

Reply:
Good question, to be honest I copied this logic from existing resources. It looks like the `interface{}` returned by `provider.ConfigureFunc` is passed into the resources:

https://github.com/signalsciences/terraform-provider-sigsci/blob/main/provider/provider.go#L69
https://github.com/signalsciences/terraform-provider-sigsci/blob/main/provider/provider.go#L95

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/helper/schema#ConfigureFunc